### PR TITLE
Correction User Group Tab Layout

### LINKED ISSUE #305: When checking user group, if there is more than 12 users, a scroll appears.
**Describe the bug**
When I'm checking a user group, if there are more then 12 users, a scroll appears instead of breaking the list to the other line or other solution.

**To Reproduce**
1. Create a user group with 13 or more users.
2. Click in the menu to see the user group created.

**Expected behavior**
I expect to do not exist a giant scroll.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by luisaborgeszup**
### Issue

When an user group had more then 12 members, an scroll bar on horizontal axis would appear. This layout wasn't matching with the In Vision design.

### Solution

In order to solve the problem, the form belonging to the tab page of the user group screen was formatted to match the In Vision design.

### Resolution

Now, the User Group module Layout is matching the In Vision design.

Invision: [User Group Screen Design](https://zup.invisionapp.com/d/main#/console/19371384/414170988/preview)

Close #305 

Signed-off-by: luisaborgeszup <luisa.borges@zup.com.br>

### CODE REVIEW

File: ui/src/modules/Groups/Tabs/Form/__tests__/Form.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render, fireEvent, wait } from 'unit-test/testUtils';
+import MutationObserver from 'mutation-observer';
+import Form from '../index';
+import userGroups from '../../../../../../stub/user-groups/mock';
```
Comment (washingtonsoares) [inline ui/src/modules/Groups/Tabs/Form/__tests__/Form.spec.tsx:21]: I think it is not good practice for a unit test to use stub files.
We must think that the stub is like a separate project, and that the test setup should guarantee the minimum possible implementation.

I've made an implementation of test fixtures, which were done to solve problems like that.

(here I just changed the test scripts for ignore fixture files) https://github.com/ZupIT/charlescd/blob/feature-metrics-circles/ui/package.json

https://github.com/ZupIT/charlescd/blob/feature-metrics-circles/ui/src/modules/Metrics/Circles/History/__tests__/fixtures.ts

If your test needs a bigger data setup, I recommend using the same approach.

File: ui/src/modules/Groups/Tabs/Form/__tests__/Form.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+test('render user group Form ', () => {
+  const { getByTestId } = render(
+    <Form
+      userGroup={userGroups.userGroups.content[0]}
+      onAddUser={jest.fn()}
+      onEdit={jest.fn()}
+    />
+  );
+  expect(
```
Comment (washingtonsoares) [inline ui/src/modules/Groups/Tabs/Form/__tests__/Form.spec.tsx:33]:
```suggestion
  );
  
  expect(
```

File: ui/src/modules/Groups/Tabs/Form/index.tsx [hunk 3]
```diff
@@ -36,9 +36,35 @@ const Form = ({ userGroup, onAddUser, onEdit }: Props) => {
     onEdit(name);
   };
 
+  const renderUsers = (userGroup: UserGroup) => {
+    let count = 0;
```
Comment (washingtonsoares) [inline ui/src/modules/Groups/Tabs/Form/index.tsx:40]: What would happen if your component was re-rendered? Perhaps the value of this variable will be lost.
What do you think of using an internal state?

File: ui/src/modules/Groups/Tabs/Form/index.tsx [hunk 4]
```diff
@@ -36,9 +36,35 @@ const Form = ({ userGroup, onAddUser, onEdit }: Props) => {
...
+    return (
+      <Styled.UserList>
+        {map(userGroup?.users, (user, index) => {
+          if (index <= 7)
```
Comment (washingtonsoares) [inline ui/src/modules/Groups/Tabs/Form/index.tsx:44]:
```suggestion
          if (index <= 7) {
```

File: ui/src/modules/Groups/Tabs/Form/__tests__/Form.spec.tsx (hunk 1 above)
Comment (washingtonsoares) [reply ui/src/modules/Groups/Tabs/Form/__tests__/Form.spec.tsx:21]: cc: @Leandroqo

File: ui/src/modules/Groups/Tabs/Form/__tests__/fixtures.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
+export const mockUserGroup1 = {
```
Comment (ifdouglas) [inline ui/src/modules/Groups/Tabs/Form/__tests__/fixtures.ts:1]: Missing license header

File: ui/src/modules/Groups/Tabs/Form/index.tsx [hunk 6]
```diff
@@ -31,14 +31,48 @@ interface Props {
 
 const Form = ({ userGroup, onAddUser, onEdit }: Props) => {
   const { register, handleSubmit } = useForm();
+  const [counter, setCounter] = useState(0);
+
+  useEffect(() => {
```
Comment (Leandroqo) [inline ui/src/modules/Groups/Tabs/Form/index.tsx:36]: I don't understand what happens in useEffect. It is interesting to break into functions that tell what they are doing.

File: ui/src/modules/Groups/Tabs/Form/__tests__/fixtures.ts (hunk 5 above)
Comment (luisaborgeszup) [reply ui/src/modules/Groups/Tabs/Form/__tests__/fixtures.ts:17]: Corrigido

File: ui/src/modules/Groups/Tabs/Form/index.tsx (hunk 6 above)
Comment (luisaborgeszup) [reply ui/src/modules/Groups/Tabs/Form/index.tsx:37]: Corrigido

**Pull request merged**